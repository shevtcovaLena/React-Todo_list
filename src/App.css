#root {
  width: 900px;
  min-width: 50%;
  margin: 0 auto;
  padding: 2rem;
  text-align: center;
}

.myTask {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
}

.inline {
  display: flex;
  justify-content: flex-start;
}

.inline input {
  margin-right: 20px;
}

.update-input {
  padding: 0.4rem;
  border-radius: 8px;
}

.card {
  padding: 2em;
}

.read-the-docs {
  color: #888;
}
