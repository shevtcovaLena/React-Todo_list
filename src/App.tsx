import { useEffect, useState } from 'react'
import './App.css'
import Form from './components/Form/Form'
import List from './components/List/List'
import { ITask } from './types/type'

// const todoInputs: ITask = {
//   title: '',
//   status:false
// }

function App() {
// const [inputs, setInputs] = useState<ITask>(todoInputs);
// const [tasks, setTasks] = useState<ITask[]>([])

// const formHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
//     setInputs((pre) => ({ ...pre, [e.target.name]: e.target.value }))
// }

// const submitHandler = async (e: React.FormEvent): Promise<void> => {
//   e.preventDefault()
//   try {
//     const response = await fetch('http://localhost:3000/api/todos', {
//       method: 'POST',
//       headers: {
//         'Content-Type':'application/json'
//       },
//       body:JSON.stringify(inputs)
//     })
//     const result = await response.json();
//     setTasks((pre) => ([...pre, result]));
//     setInputs(() => todoInputs)
//   } catch (error) {
//     console.log(error)
//   }
// }

// const delHandler = async (id: number) => {
//   try {
//     const response = await fetch(`http://localhost:3000/api/todos/${id}`, {
//       method: 'DELETE',
//   })
//     if(response.status === 200) {
//       setTasks((pre) => pre.filter((el) => el.id !== id))
//     }
//   } catch (error) {
//     console.log(error)
//   }
// }

// const checkHandler = async (id: number) => {
//   const myTask : ITask | undefined = tasks.find((el) => el.id === id);
//   // console.log(myTask)
//   if (myTask) {    
//     try {
//       const response = await fetch(`http://localhost:3000/api/todos/${id}`, {
//         method: 'PATCH',
//         headers: {
//           'Content-Type':'application/json'
//         },
//         body:JSON.stringify({ status: !myTask.status })
//       })
//       const result = await response.json()
//       if(response.status === 200) {
//         setTasks((pre) => pre.map((el) => {
//           if (el.id === id ) {
//             el.status = result.status 
//             return el
//           } 
//           return el
//         }))
//       }
//     } catch (error) {
//       console.log(error)
//     }
//   }
// }



// const updateHandler = async (id: number, str: string): Promise<void> => {   

//     try {
//       const response = await fetch(`http://localhost:3000/api/todos/${id}`, {
//         method: 'PATCH',
//         headers: {
//           'Content-Type':'application/json'
//         },
//         body:JSON.stringify({ title: str })
//       })
//       const result = await response.json()
//       if(response.status === 200) {
//         setTasks((pre) => pre.map((el) => {
//           if (el.id === id ) {
//             el.title = result.title
//             return el
//           } 
//           return el
//         }))
//       }
//     } catch (error) {
//       console.log(error)
//     }
// }

// useEffect(():void => {
//   ( async function (): Promise<void> {
//     try {
//       const response = await fetch('http://localhost:3000/api/todos');
//       const result = await response.json();
//       setTasks(() => ([...result]))
//     } catch (error) {
//       console.log(error)
//     }
//   })()
// }, [])
  return (
    <>
    {/* <Form submitHandler={ submitHandler } formHandler={ formHandler } inputs = { inputs }/>
    <List tasks={tasks} delHandler={ delHandler } checkHandler={ checkHandler }  updateHandler={updateHandler}/> */}
    <Form/>
    <List/>
    </>
  )
}

export default App
