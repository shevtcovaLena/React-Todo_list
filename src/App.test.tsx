// import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import App from './App';

beforeAll(() => {
  window.matchMedia = window.matchMedia || function() {
    return {
      matches: false,
      addListener: function() {},
      removeListener: function() {}
    };
  };
});

test('renders learn react link', () => {
  render(<App />);
  const linkElement = screen.getByText(/добавить/i);
  expect(linkElement).toBeInTheDocument();
});