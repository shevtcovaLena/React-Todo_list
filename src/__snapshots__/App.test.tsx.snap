// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correct ListCard render > button 1`] = `
<form
  class="ant-form ant-form-horizontal css-dev-only-do-not-override-1m81x91"
  data-testid="task-form"
  style="margin: 20px auto;"
>
  <div
    class="ant-form-item css-dev-only-do-not-override-1m81x91"
  >
    <div
      class="ant-row ant-form-item-row css-dev-only-do-not-override-1m81x91"
    >
      <div
        class="ant-col ant-form-item-control css-dev-only-do-not-override-1m81x91"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <span
              class="ant-input-affix-wrapper ant-input-affix-wrapper-lg css-dev-only-do-not-override-1m81x91 ant-input-outlined"
              style="background-color: white; height: 3rem;"
            >
              <span
                class="ant-input-prefix"
              >
                <span
                  aria-label="snippets"
                  class="anticon anticon-snippets"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="snippets"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M832 112H724V72c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v40H500V72c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v40H320c-17.7 0-32 14.3-32 32v120h-96c-17.7 0-32 14.3-32 32v632c0 17.7 14.3 32 32 32h512c17.7 0 32-14.3 32-32v-96h96c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM664 888H232V336h218v174c0 22.1 17.9 40 40 40h174v338zm0-402H514V336h.2L664 485.8v.2zm128 274h-56V456L544 264H360v-80h68v32c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-32h152v32c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-32h68v576z"
                    />
                  </svg>
                </span>
              </span>
              <input
                class="ant-input ant-input-lg css-dev-only-do-not-override-1m81x91"
                id="title"
                placeholder="Добавьте заметку"
                type="text"
                value=""
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <button
    class="ant-btn css-dev-only-do-not-override-1m81x91 ant-btn-primary"
    type="button"
  >
    <span>
      Добавить
    </span>
  </button>
</form>
`;

exports[`correct ListCard render button 1`] = `
<form
  class="ant-form ant-form-horizontal css-dev-only-do-not-override-1m81x91"
  data-testid="task-form"
  style="margin: 20px auto;"
>
  <div
    class="ant-form-item css-dev-only-do-not-override-1m81x91"
  >
    <div
      class="ant-row ant-form-item-row css-dev-only-do-not-override-1m81x91"
    >
      <div
        class="ant-col ant-form-item-control css-dev-only-do-not-override-1m81x91"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <span
              class="ant-input-affix-wrapper ant-input-affix-wrapper-lg css-dev-only-do-not-override-1m81x91 ant-input-outlined"
              style="background-color: white; height: 3rem;"
            >
              <span
                class="ant-input-prefix"
              >
                <span
                  aria-label="snippets"
                  class="anticon anticon-snippets"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="snippets"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M832 112H724V72c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v40H500V72c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v40H320c-17.7 0-32 14.3-32 32v120h-96c-17.7 0-32 14.3-32 32v632c0 17.7 14.3 32 32 32h512c17.7 0 32-14.3 32-32v-96h96c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM664 888H232V336h218v174c0 22.1 17.9 40 40 40h174v338zm0-402H514V336h.2L664 485.8v.2zm128 274h-56V456L544 264H360v-80h68v32c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-32h152v32c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-32h68v576z"
                    />
                  </svg>
                </span>
              </span>
              <input
                class="ant-input ant-input-lg css-dev-only-do-not-override-1m81x91"
                id="title"
                placeholder="Добавьте заметку"
                type="text"
                value=""
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <button
    class="ant-btn css-dev-only-do-not-override-1m81x91 ant-btn-primary"
    type="button"
  >
    <span>
      Добавить
    </span>
  </button>
</form>
`;
